src/Client.ts

```diff
@@ -532,3 +620,38 @@ async function getUserContactFromNetwork(
   }
   return undefined
 }
+
+// retrieve a list of key bundles given a list of user addresses
+async function getUserContactsFromNetwork(
+  apiClient: ApiClient,
+  peerAddresses: string[]
+): Promise<(PublicKeyBundle | SignedPublicKeyBundle | undefined)[]> {
+  const userContactTopics = peerAddresses.map(buildUserContactTopic)
+  const topicToEnvelopes = await apiClient.batchQuery(
+    userContactTopics.map((topic) => ({
+      contentTopics: [topic],
+      pageSize: 5,
+      direction: SortDirection.SORT_DIRECTION_DESCENDING,
+    }))
+  )
+
+  // Transform topicToEnvelopes into a list of PublicKeyBundles or undefined
+  // by going through each message and attempting to decode
+  return Promise.all(
+    peerAddresses.map(async (address: string, index: number) => {
+      const envelopes = topicToEnvelopes[index]
+      if (!envelopes) {
+        return undefined
+      }
+      for (const env of envelopes) {
+        if (!env.message) continue
+        const keyBundle = decodeContactBundle(b64Decode(env.message.toString()))
```

Comment:
Think we will want to wrap this in a try/catch. A single undecodable bundle would bubble up and cause the entire `Promise.all` to fail

Reply:
Good catch, I'll make the change and see if it's easy to add a test for it too